pkg/authz/permission_list_app.go

```diff
@@ -0,0 +1,76 @@
+package authz
+
+func init() {
+	AppPermissionList = append(AppPermissionList, Permission{
+		APIGroups:     []string{"primaza.io"},
+		Resources:     []string{"servicebindings", "serviceclaims", "servicecatalogs"},
+		ResourceNames: []string{},
+		Namespace:     "system",
+		Name:          "primaza:app:manager",
+		Verbs:         []string{"get", "list", "watch", "update", "patch", "delete", "deletecollection"},
+	})
+	AppPermissionList = append(AppPermissionList, Permission{
+		APIGroups:     []string{"primaza.io"},
+		Resources:     []string{"serviceclaims/status"},
+		ResourceNames: []string{},
+		Namespace:     "system",
+		Name:          "primaza:app:manager",
+		Verbs:         []string{"get", "list", "watch", "update"},
+	})
```

Comment:
```suggestion
var AppPermissionList = []Permission{
	{
		APIGroups:     []string{"primaza.io"},
		Resources:     []string{"servicebindings", "serviceclaims", "servicecatalogs"},
		ResourceNames: []string{},
		Namespace:     "system",
		Name:          "primaza:app:manager",
		Verbs:         []string{"get", "list", "watch", "update", "patch", "delete", "deletecollection"},
	},
	{
		APIGroups:     []string{"primaza.io"},
		Resources:     []string{"serviceclaims/status"},
		ResourceNames: []string{},
		Namespace:     "system",
		Name:          "primaza:app:manager",
		Verbs:         []string{"get", "list", "watch", "update"},
	},
```

Reply:
I prefer to retain the current approach of initializing `AppPermissionList`.  Few reasons:

1. There is a clear separation between the variable declaration (`pkg/authz/permissions.go`) and initialization.
2. Since `init` is called once during runtime, there is no performance issue.
3. The `pkg/authz/permission_list_app.go` is a generated code. In fact, we need not commit this script 
   in the source tree. The program builds even without the generated files.
4. I don't see any advantage in declaring and initializing the `AppPermissionList` in the generated code. In fact,
   the program won't build without the generated code, if we take this approach.

Reply:
1. This is a PRO for sure, but to me this not justifies the CONS that follows
2. `init` is called when the module is loaded. This side-effect on module import may be really annoying and hard to test, reduce readability and increase complexity (cfr. https://github.com/leighmcculloch/gochecknoinits)
3. I'd suggest to commit the generated code. This will give the repo reader a full understanding of what's really happening.
4. I'm expecting the code to not build without generating code and checks on pipeline to be sure to have the latest generated code. This way what you see in the repo is matching exactly what it's built for a given version of the binary

Reply:
Thanks for the detailed explanation. I agree with your points. Let me make the necessary changes.

Reply:
Done